post-docs/add-back-references.py

```diff
@@ -14,16 +14,34 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
-
+import enum
+import logging
 import os
+import sys
 from urllib.request import urlopen
 import semver
 
-docs_link = "https://raw.githubusercontent.com/apache/airflow/main/docs/apache-airflow/redirects.txt"
+airflow_redirects_link = "https://raw.githubusercontent.com/apache/airflow/main/docs/apache-airflow/redirects.txt"
+helm_redirects_link = "https://raw.githubusercontent.com/apache/airflow/main/docs/helm-chart/redirects.txt"
+providers_redirect_link = "populate-this"
+
 docs_archive_path = "../docs-archive"
-apache_airflow_path = docs_archive_path + "/apache-airflow"
-stable_version_path = apache_airflow_path + "/stable.txt"
-new_docs_version = "2.5.1"
+airflow_docs_path = docs_archive_path + "/apache-airflow"
+helm_docs_path = docs_archive_path + "/helm-chart"
+providers_docs_path = docs_archive_path + "/apache-airflow-providers"
+
+# version where changes were introduced in docs structure
+new_airflow_docs_version = "2.5.1"
```

Comment:
Question: Can we do it regardless for latest versions ?  I believe (correct me if I am wrong please) that when you get the "latest" version, and there are no redirects we could check if the file that we want to create redirect from exists already and skip if it does - this way we will not override "target" files, but we will be able to re-run it at any time from latest version, making it effectively idempotent.

Smth like: 

```

def create_back_reference_html(back_ref_url, path):
    content = get_redirect_content(back_ref_url)
   if Path(path).exists(): # <-- check if the target file exists already
      return 
    # Creating an HTML file
    with open(path, "w") as f:
        f.write(content)
```

Reply:
Very nice point, let me try out this approach.

Reply:
Fetching and filtering by version seems like an additional overhead which we should try to avoid as much as possible.